service/controller/machinepool/handler/migration/machinepool_test.go

```diff
@@ -0,0 +1,294 @@
+package migration
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	oldcapiexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+
+	"github.com/giantswarm/azure-operator/v5/service/unittest"
+)
+
+func TestEnsureCreatedMachinePoolIsCorrect(t *testing.T) {
+	testCases := []struct {
+		name               string
+		oldMachinePoolFile string
+		newMachinePoolFile string
+		expected           expected
+	}{
+		{
+			name:               "case 0: New MachinePool references not updated",
+			oldMachinePoolFile: "old_machinepool.yaml",
+			newMachinePoolFile: "new_machinepool_references_not_updated.yaml",
+		},
+		{
+			name:               "case 1: New MachinePool status not updated",
+			oldMachinePoolFile: "old_machinepool.yaml",
+			newMachinePoolFile: "new_machinepool_status_not_updated.yaml",
+		},
+		{
+			name:               "case 2: New MachinePool fully created",
```

Comment:
New MachinePool is fully created (Metadata, Spec and Status), so new handler will not do anything.